Unable to build on Mac OS 10.15

## Bug Report

**Describe the bug**
Unable to build fluentbit from source on MacOS 10.15 

**To Reproduce**

Getting this error.

```
[  3%] Built target flb-plugin-proxy-go
[  3%] Performing build step for 'luajit'
==== Building LuaJIT 2.1.0-beta3 ====
HOSTCC    host/minilua.o
host/minilua.c:34:10: fatal error: 'math.h' file not found
#include <math.h>
         ^~~~~~~~
1 error generated.
make[4]: *** [host/minilua.o] Error 1
make[3]: *** [default] Error 2
make[2]: *** [luajit-prefix/src/luajit-stamp/luajit-build] Error 2
make[1]: *** [CMakeFiles/luajit.dir/all] Error 2
make: *** [all] Error 2
```

**Expected behavior**
Should be build without any issues

**Your Environment**

* Version used: 1.5 latest (on master)
* Configuration: 
* Environment name and version (e.g. Kubernetes? What version?): Local Setup
* Server type and version:
* Operating System and version: Mac OS Catalina 10.15.2
* Filters and plugins: No plugins

cc @edsiper

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ if (MSVC)
   # Use add_compile_options() to set /MT since Visual Studio
   # Generator does not notice /MT in CMAKE_C_FLAGS.
   add_compile_options(/MT)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fno-common")
 else()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 endif()

--- a/cmake/luajit.cmake
+++ b/cmake/luajit.cmake
@@ -11,6 +11,10 @@ add_library(libluajit STATIC IMPORTED GLOBAL)
 set(LUAJIT_SRC ${CMAKE_CURRENT_SOURCE_DIR}/lib/LuaJIT-2.1.0-beta3)
 set(LUAJIT_DEST ${CMAKE_CURRENT_BINARY_DIR})
 
+if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(CFLAGS "${CFLAGS} -isysroot ${CMAKE_OSX_SYSROOT}")
+endif()
+
 # luajit (UNIX)
 # =============
 ExternalProject_Add(luajit
@@ -18,7 +22,7 @@ ExternalProject_Add(luajit
   EXCLUDE_FROM_ALL TRUE
   SOURCE_DIR ${LUAJIT_SRC}
   CONFIGURE_COMMAND ./configure
-  BUILD_COMMAND $(MAKE) CC=${CMAKE_C_COMPILER} BUILD_MODE=static XCFLAGS="-fPIC"
+  BUILD_COMMAND $(MAKE) CC=${CMAKE_C_COMPILER} CFLAGS=${CFLAGS} BUILD_MODE=static "XCFLAGS=-fPIC"
   INSTALL_COMMAND cp src/libluajit.a "${LUAJIT_DEST}/lib/libluajit.a")
 
 # luajit (Windows)

